src/main/java/com/alvarium/streams/StreamConfig.java

```diff
@@ -0,0 +1,9 @@
+package com.alvarium.streams;
+
+
+/**
+ * Marker interface for different stream config classes
+ */
+public interface StreamConfig {
+
+ }
```

Comment:
My initial impression is that this can be removed and the instance of the config can be treated as an `Object`. What good is an interface without any members? I imagine you're trying to duplicate the functionality from the Go SDK where the type of the config isn't known until runtime.

Reply:
You are correct in assuming that it's there because the type of the config isn't known in compile time, but it also seemed to us that using the marker interface design pattern to create a domain for the different stream configuration classes might better than using `Object` as this way will narrow down the possible types that can be used.